### PR TITLE
e2e: get with sort order, target

### PR DESCRIPTION
**Pull request by gyuho**

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @heyitsanthony @xiang90 

Thanks!

**Comment by heyitsanthony** [comment]
This works but on the other hand it's turning out to be more complicated and brittle than it seems this sort of thing should be

I tried writing the get code as a test table instead, what do you think?
```go
func testCtlV3Get(cx ctlCtx) {
        defer close(cx.errc)

        kvs := []kv{{"key1", "val1"}, {"key2", "val2"}, {"key3", "val3"}}
        revkvs := []kv{{"key3", "val3"}, {"key2", "val2"}, {"key1", "val1"}}

        tests := []struct {
                args []string

                wkv []kv
        }{
                {[]string{"key1"}, []kv{{"key1", "val1"}}},
                {[]string{"key", "--prefix"}, kvs},
                {[]string{"key", "--prefix", "--limit=2"}, kvs[:2]},
                {[]string{"key", "--prefix", "--order=ASCEND", "--sort-by=MODIFY"}, kvs},
                {[]string{"key", "--prefix", "--order=ASCEND", "--sort-by=VERSION"}, kvs},
                {[]string{"key", "--prefix", "--order=DESCEND", "--sort-by=CREATE"}, revkvs},
                {[]string{"key", "--prefix", "--order=DESCEND", "--sort-by=KEY"}, revkvs},
        }

        for i := range kvs {
                if err := ctlV3Put(cx, kvs[i].key, kvs[i].val); err != nil {
                        cx.t.Fatalf("testCtlV3Get ctlV3Put error (%v)", err)
                }
        }

        for i, tt := range tests {
                if err := ctlV3Get(cx, tt.args, tt.wkv...); err != nil {
                        cx.t.Errorf("#%v (%v)", i, err)
                }
        }
}
```

**Comment by gyuho** [comment]
@heyitsanthony Right. I think your table is more clearer about what it is testing. I will fix this commit using your tables. Thanks!

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Fixed as you suggested. Looks better now.

Thanks!

File: e2e/etcdctlv3_test.go [hunk 1]
```diff
@@ -103,6 +100,9 @@ type ctlCtx struct {
 	interactive   bool
 	watchRevision int
 
+	sortOrder  string
+	sortTarget string
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: delete these? limit too

File: e2e/etcdctlv3_test.go [hunk 2]
```diff
@@ -138,6 +138,14 @@ func withWatchRevision(rev int) ctlOption {
 	return func(cx *ctlCtx) { cx.watchRevision = rev }
 }
 
+func withSortOrder(od string) ctlOption {
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: delete?

File: e2e/etcdctlv3_test.go [hunk 3]
```diff
@@ -359,19 +374,9 @@ type kv struct {
 	key, val string
 }
 
-func ctlV3Get(cx ctlCtx, key string, kvs ...kv) error {
-	cmdArgs := append(ctlV3PrefixArgs(cx.epc, cx.dialTimeout), "get", key)
-	if !cx.quorum {
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: still needs this

File: e2e/etcdctlv3_test.go [hunk 4]
```diff
@@ -359,19 +374,9 @@ type kv struct {
...
-	}
-	if cx.prefix {
-		cmdArgs = append(cmdArgs, "--prefix")
-	}
-	if cx.limit > 0 {
-		cmdArgs = append(cmdArgs, "--limit", strconv.Itoa(cx.limit))
-	}
-	// TODO: configure order, sort-by
-
+func ctlV3Get(cx ctlCtx, args []string, kvs ...kv) error {
```
Comment (heyitsanthony) [inline e2e/etcdctlv3_test.go]: should include the dial timeout error test?

File: e2e/etcdctlv3_test.go (hunk 3 above)
Comment (gyuho) [inline e2e/etcdctlv3_test.go]: right.

File: e2e/etcdctlv3_test.go (hunk 4 above)
Comment (gyuho) [inline e2e/etcdctlv3_test.go]: Ok will add.

**Comment by gyuho** [comment]
All addressed. PTAL. Thanks!

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
Thanks. Will merge after green lights.

**Pull request merged**